app/main.py

```diff
@@ -1,13 +1,34 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
-
+    return {
+        "line": error.get("line_number"),
+        "column": error.get("column_number"),
+        "message": error.get("text"),
+        "name": error.get("code"),
+        "source": "flake8",
+    }
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
-
+    return {
+        "path": file_path,
+        "errors": [format_linter_error(error) for error in errors],
+        "status": "failed" if errors else "passed"
+    }
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "path": file_path,
+            "errors": [
+                {
+                    "line": error["line_number"],
+                    "column": error["column_number"],
+                    "message": error["text"],
+                    "name": error["code"],
+                    "source": "flake8",
+                }
+                for error in errors
+            ],
+            "status": "failed" if errors else "passed"
+        }
+        for file_path, errors in linter_report.items()
+    ]
```

Comment:
The logic for formatting each file's errors in `format_linter_report` is redundant with the `format_single_linter_file` function. Consider using `format_single_linter_file(file_path, errors)` within this list comprehension to avoid code duplication and improve maintainability.